pkg/controller/operators/olm/operatorgroup.go

```diff
@@ -981,59 +996,88 @@ func (a *Operator) updateNamespaceList(op *operatorsv1.OperatorGroup) ([]string,
 	return namespaceList, nil
 }
 
-func (a *Operator) ensureOpGroupClusterRole(op *operatorsv1.OperatorGroup, suffix string, apis cache.APISet) error {
-	clusterRole := &rbacv1.ClusterRole{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:   strings.Join([]string{op.GetName(), suffix}, "-"),
-			Labels: map[string]string{install.OLMManagedLabelKey: install.OLMManagedLabelValue},
-		},
+func (a *Operator) getClusterRoleName(op *operatorsv1.OperatorGroup, roleType string) (string, error) {
```

Comment:
note: in the degenerate case where there's 0 characters available for the `op.GetName()` the role name does not collide with others since roleType is in the hash

Reply:
Unit tests would help!

Reply:
But... mathematically impossible for `nameLimit` to be `<0`. The only thing not a constant value is the roleType.